### PR TITLE
update externalcluster response

### LINKED ISSUE #10391: Differentiate Imported and Created External Clusters
### Description of the feature you would like to add / User story

As a developer
I would like to Differentiate Imported and Created External Clusters using a label
in order to Differentiate between the two

### Solution details

- Requirement 1
- Requirement 2
- Requirement 3

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What does this PR do / Why do we need it**: 
- update "get external cluster md" response to return details to support KKP UI
- Adds a label `IsImported` to distinguish between imported and created external clusters

**Does this PR close any issues?**:
Fixes #10391

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: pkg/api/v2/types.go [hunk 1]
```diff
@@ -629,14 +669,14 @@ type AKSCloudSpec struct {
...
+	// The identity that created the resource.
+	CreatedBy *string `json:"createdBy,omitempty"`
 	// KubernetesVersion - When you upgrade a supported AKS cluster, Kubernetes minor versions cannot be skipped. All upgrades must be performed sequentially by major version number. For example, upgrades between 1.14.x -> 1.15.x or 1.15.x -> 1.16.x are allowed, however 1.14.x -> 1.16.x is not allowed. See [upgrading an AKS cluster](https://docs.microsoft.com/azure/aks/upgrade-cluster) for more details.
 	KubernetesVersion string `json:"kubernetesVersion"`
-	// NodeResourceGroup - The name of the resource group containing agent pool nodes.
-	NodeResourceGroup string `json:"nodeResourceGroup,omitempty"`
 	// EnableRBAC - Whether Kubernetes Role-Based Access Control Enabled.
 	EnableRBAC bool `json:"enableRBAC,omitempty"`
-	// ManagedAAD - Whether The Azure Active Directory configuration Enabled.
-	ManagedAAD bool `json:"managedAAD,omitempty"`
```
Comment (lsviben) [inline pkg/api/v2/types.go:639]: why are we removing the ManagedAAD and NodeResourceGroup?

Comment (imharshita) [reply pkg/api/v2/types.go:639]: i just removed it as it seemed to me like an non critical info and it can crowd the cluster view

Comment (lsviben) [reply pkg/api/v2/types.go:639]: i see. This was not released in the previous release right? so we are not changing the API? If not, then ok we can  remove it :)

File: pkg/api/v2/types.go [hunk 2]
```diff
@@ -603,17 +605,55 @@ type EKSClusterSpec struct {
...
+	Tags map[string]*string `json:"tags,omitempty"`
 
 	// The Amazon Resource Name (ARN) of the IAM role that provides permissions
 	// for the Kubernetes control plane to make calls to AWS API operations on your
 	// behalf. For more information, see Amazon EKS Service IAM Role (https://docs.aws.amazon.com/eks/latest/userguide/service_IAM_role.html)
 	// in the Amazon EKS User Guide .
 	//
 	// RoleArn is a required field
-	RoleArn string `json:"roleArn" required:"true"`
+	RoleArn string `json:"roleArn,omitempty" required:"true"`
```
Comment (pkprzekwas) [inline pkg/api/v2/types.go:630]: Is it fine to have `omitempty` and required set to `true` at the same time?

**Comment by imharshita** [comment]
> 

Yes, we can think about it. But for now there is no solid way to check if cluster is imported or created after the creation.
I can create a issue to discuss this

File: pkg/api/v2/types.go (hunk 2 above)
Comment (imharshita) [reply pkg/api/v2/types.go:630]: yeah, i know its confusing.
the reason is "required" for creation/POST request but `omitempty` for get response

File: pkg/api/v2/types.go [hunk 3]
```diff
@@ -603,17 +605,55 @@ type EKSClusterSpec struct {
...
+}
+
+// The Kubernetes network configuration for the cluster. The response contains
+// a value for serviceIpv6Cidr or serviceIpv4Cidr, but not both.
+type KubernetesNetworkConfigResponse struct {
```
Comment (pkprzekwas) [inline pkg/api/v2/types.go]: You mention Amazon CNI  several times in the doc and looks like only EKS Cluster Spec uses it. Is this type EKS specific? If so maybe we should relate that in the name.

File: pkg/api/v2/types.go (hunk 2 above)
Comment (imharshita) [reply pkg/api/v2/types.go:630]: wdyt? i can add a comment here to explain, or just remove the `omitempty`

File: pkg/handler/v2/external_cluster/eks.go [hunk 4]
```diff
@@ -859,14 +871,22 @@ func getEKSClusterDetails(ctx context.Context, apiCluster *apiv2.ExternalCluster
...
+			IpFamily:        eksCluster.KubernetesNetworkConfig.IpFamily,
+			ServiceIpv4Cidr: eksCluster.KubernetesNetworkConfig.ServiceIpv4Cidr,
+			ServiceIpv6Cidr: eksCluster.KubernetesNetworkConfig.ServiceIpv6Cidr,
+		}
 	}
 
 	if eksCluster.ResourcesVpcConfig != nil {
 		clusterSpec.ResourcesVpcConfig = apiv2.VpcConfigRequest{
-			SecurityGroupIds: eksCluster.ResourcesVpcConfig.SecurityGroupIds,
-			SubnetIds:        eksCluster.ResourcesVpcConfig.SubnetIds,
```
Comment (pkprzekwas) [inline pkg/handler/v2/external_cluster/eks.go:869]: We can add `SecurityGroupIds ` , it's not a secret.

**Pull request merged**